app/models/user.rb

```diff
@@ -199,6 +201,28 @@ def username_required?
   def email_required?
     !erased?
   end
+  
+  def has_official_email?
+    domain = Setting.value_for 'email_domain_for_officials'
+    return false if !email or !domain or domain.length == 0
```

Comment:
Please use the symbolic operator `||` instead of `or` to avoid confusion and unexpected behaviour here.